api/envoy/config/filter/http/cache/v2/cache.proto

```diff
@@ -0,0 +1,63 @@
+syntax = "proto3";
+
+package envoy.config.filter.http.cache.v2;
+
+import "validate/validate.proto";
+
+option java_package = "io.envoyproxy.envoy.config.filter.http.cache.v2";
+option java_outer_classname = "CacheProto";
+option java_multiple_files = true;
+
+// [#protodoc-title: HTTP Cache Filter]
+// Pluggable HTTP Cache :ref:`overview <config_http_filters_cache>`.
+// [#extension: envoy.filters.http.cache]
+message Cache {
+  // List of query params.
+  message QueryParams {
+    repeated string query_string_keys = 1;
+  }
+
+  // Modifies cache key creation by restricting which parts of the URL are included.
+  message KeyCreatorParams {
+    // If true, exclude the URL scheme from the cache key. Set to true if your origins always
+    // produce the same response for http and https requests.
+    bool exclude_scheme = 1;
+
+    // If true, exclude the host from the cache key. Set to true if your origins' responses don't
+    // ever depend on host.
+    bool exclude_host = 2;
+
+    oneof query_string_keys {
+      // If set, only query params in this list are included in the cache
+      // key. Any other query params will not affect cache lookup. If set to an
+      // empty list, the query string will be entirely ignored in cache lookups.
+      QueryParams query_string_keys_included = 3;
+
+      // If set, query params in this list are excluded from the cache key, and
+      // will not affect cache lookup. Setting this to an empty list is
+      // equivalent to not setting it (but is less efficient).
+      QueryParams query_string_keys_excluded = 4;
+    }
+  }
+
+  // Name of cache implementation to use.
+  string name = 1 [(validate.rules).string = {min_bytes: 1}];
```

Comment:
Is this a 1st class Envoy extension point? Do the cache implementation need their own `typed_config` etc?

Reply:
In general, would be good to understand which configuration is general and applicable to all implementations, and which is per-implementation.

Reply:
Everything here is general, and will be used by the cache filter. My intent is to add an opaque implementation-specific config in a future PR, once I know the proper Envoy way to do that.

Reply:
Ack.